index.html

```diff
@@ -992,24 +993,37 @@ <h3>Overview</h3>
       <h3>Algorithm</h3>
 
       <p>This algorithm specifies how a new <a>active context</a> is updated
-        with a <a>local context</a>. The algorithm takes three input variables:
-        an <var>active context</var>, a <var>local context</var>, and an  <a>array</a>
-        <var>remote contexts</var> which is used to detect cyclical context inclusions.
-        If <var>remote contexts</var> is not passed, it is initialized to an empty
-        <a>array</a>.</p>
+        with a <a>local context</a>. The algorithm takes two required
+        <span class="changed">and two optional</span>
+        input variables:
+        an <var>active context</var> and a <var>local context</var>.
+        <span class="changed">The optional inputs are <var>from term</var></span>,
+        and an <a>array</a>
+        <var>remote contexts</var> which is used to detect cyclical context inclusions,
+        which defaults to a new empty <a>array</a>.</p>
 
       <ol>
         <li>Initialize <var>result</var> to the result of cloning
           <var>active context</var>.</li>
+        <li>Initialize <var>current seal</var> to the value of <a>context id</a>
+          from the term definition for <var>from term</var>,
+          if such a term definition exists.</li>
         <li>If <var>local context</var> is not an <a>array</a>,
           set it to an <a>array</a> containing only
           <var>local context</var>.</li>
         <li>
           For each item <var>context</var> in <var>local context</var>:
           <ol>
-            <li>If <var>context</var> is <code>null</code>, set <var>result</var> to a
+            <li class="changed">Initialize <var>context id</var> to a unique identifier.
+              <span class="note"><var>context id</var> is used to identify sealed terms
+                created from the same context.</span></li>
+            <li>If <var>context</var> is <code>null</code>
+              <span class="changed">and <var>current seal</var> is <code>null</code></span>,
+              set <var>result</var> to a
               newly-initialized <var>active context</var> and continue with the
               next <var>context</var>.
+              <span class="changed">If <var>current seal</var> is <code>null</code>,
+                processors SHOULD issue a warning that an attempt to clear a sealed context was attempted.</span>
```

Comment:
replace "was attempted" by "was made". "an attempt was attempted" is a bit ugly :)